src/Evolution/DgSubcell/SetInterpolators.hpp

```diff
@@ -0,0 +1,237 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/functional/hash.hpp>
+#include <cstddef>
+#include <optional>
+#include <utility>
+
+#include "DataStructures/FixedHashMap.hpp"
+#include "DataStructures/Tensor/ExtractPoint.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Creators/Tags/Domain.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/Structure/BlockId.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/SliceTensor.hpp"
+#include "Evolution/DgSubcell/Tags/Interpolators.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
+#include "Evolution/DgSubcell/Tags/Reconstructor.hpp"
+#include "NumericalAlgorithms/Interpolation/IrregularInterpolant.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace evolution::dg::subcell {
+/*!
+ * \brief Sets the `intrp::IrregularInterpolant`s for interpolating to ghost
+ * zone data at block boundaries.
+ *
+ * The DG to FD interpolants are at full order of the DG grid. The FD to FD
+ * interpolant is piecewise linear with no support for neighboring
+ * elements. We will want to use high-order slope-limited FD interpolation in
+ * the future, but that requires neighbor communication. A slightly simpler
+ * approach would be to use high-order Lagrange interpolation, which still
+ * requires neighbor communication but does not require any additional changes
+ * to the reconstruction routines to work on non-uniform grids. This is what
+ * the Multipatch-MHD code does, relying on the slope limiting from the ghost
+ * zones to remove oscillations. I (Nils Deppe) am not sure I love that, but
+ * it's worth a try since it should be pretty easy to do.
+ */
+template <size_t Dim>
+struct SetInterpolators {
+  using return_tags = tmpl::list<
+      evolution::dg::subcell::Tags::InterpolatorsFromFdToNeighborFd<Dim>,
+      evolution::dg::subcell::Tags::InterpolatorsFromDgToNeighborFd<Dim>,
+      evolution::dg::subcell::Tags::InterpolatorsFromNeighborDgToFd<Dim>>;
+  using argument_tags =
+      tmpl::list<::domain::Tags::Element<Dim>, ::domain::Tags::Domain<Dim>,
+                 domain::Tags::Mesh<Dim>, domain::Tags::Mesh<Dim>,
```

Comment:
why are you passing in the same tag twice?

Reply:
In the future, when we have non-uniform h/p-refinement this needs to take neighbor meshes. That should be documented. I'll add a `warning` in the Dox and also a C++ comment in front of the arguments